jl4-core/src/L4/TypeCheck/Types.hs

```diff
@@ -22,6 +23,7 @@ type Substitution = Map Int (Type' Resolved)
 data CheckEntity =
     KnownType Kind [Resolved] (TypeDecl Resolved)
   | KnownTerm (Type' Resolved) TermKind
+  | KnownSection (Section Resolved)
```

Comment:
I am not sure I'd expect `Section Resolved` here.

I think I would expect something like `[CheckInfo]` here?